akka-persistence-query/src/main/scala/akka/persistence/query/internal/QuerySerializer.scala

```diff
@@ -237,4 +244,39 @@ import akka.util.ccompat.JavaConverters._
     str.toString
   }
 
+  private def timestampOffsetBySliceFromStorageRepresentation(str: String): TimestampOffsetBySlice = {
+    try {
+      val offsets =
+        if (str.isEmpty) Map.empty[Int, TimestampOffset]
+        else
+          str
+            .split(timestampOffsetBySliceSeparator)
+            .map { entry =>
+              val sliceSeparator = entry.indexOf(timestampOffsetSeparator)
+              val slice = entry.substring(0, sliceSeparator).toInt
+              val offset = timestampOffsetFromStorageRepresentation(entry.substring(sliceSeparator + 1))
+              slice -> offset
+            }
+            .toMap
+      TimestampOffsetBySlice(offsets)
+    } catch {
+      case NonFatal(e) =>
+        throw new IllegalArgumentException(s"Unexpected serialized TimestampOffsetBySlice format [$str].", e)
+    }
+  }
+
+  private def timestampOffsetBySliceToStorageRepresentation(offsetBySlice: TimestampOffsetBySlice): String = {
+    val str = new java.lang.StringBuilder
+    if (offsetBySlice.offsets.nonEmpty) {
+      var first = true
+      offsetBySlice.offsets.foreach {
```

Comment:
Seen map in timestamp offset is sorted by pid first. Could do something similar here, and sort by slice if it's useful to have it in stable order.